include/bluetooth/audio/audio.h

```diff
@@ -1073,6 +1114,40 @@ struct bt_audio_unicast_server_cb {
 	 */
 	int (*publish_capability)(struct bt_conn *conn, uint8_t type,
 				  uint8_t index, struct bt_codec *const codec);
+
+	/** @brief Publish location callback
+	 *
+	 *  Publish location callback is called whenever a remote client
+	 *  requests to read the Published Audio Capabilities (PAC) location,
+	 *  or if the location needs to be notified.
+	 *
+	 *  @param[in]  conn      The connection that requests the location.
+	 *                        Will be NULL if the location is requested
+	 *                        for sending a notification, as a result of
+	 *                        callling
+	 *                        bt_audio_unicast_server_location_changed().
+	 *  @param[in]  type      Type of the endpoint.
+	 *  @param[out] location  Pointer to the location that needs to be set.
+	 *
+	 *  @return 0 in case of success or negative value in case of error.
+	 */
+	int (*publish_location)(struct bt_conn *conn,
+				enum bt_audio_pac_type type,
+				enum bt_audio_location *location);
+
+	/** @brief Write location callback
+	 *
+	 *  Write location callback is called whenever a remote client
+	 *  requests to write the Published Audio Capabilities (PAC) location.
+	 *
+	 *  @param conn      The connection that requests the write.
+	 *  @param type      Type of the endpoint.
+	 *  @param location  The location being written.
+	 *
+	 *  @return 0 in case of success or negative value in case of error.
+	 */
+	int (*write_location)(struct bt_conn *conn, enum bt_audio_pac_type type,
+			      enum bt_audio_location location);
```

Comment:
Why the location cannot be stored internally?

Reply:
It can, however PACS was (is) designed to be stateless, i.e. all the values are located in an upper layer. 

It would have been easy to store the locaiton in PACS and still keep e.g. the capabilities in an upper layer, but I did not want to mix-match where the service data/states were stored, so used the same approach for the location as we have for the other values in PACS. 

This of course is a more complex implementation and possibly erroneous as the notifications when the location is changed is no longer ensured by the stack. 

We have yet to decided what final approach we want to take for the LE Audio services, as both approaches have their pros and cons.

Reply:
I believe an application can change the value run-time - hence I think this implementation is how it should be.
But reading the description again I think:
	 *  Write location callback is called whenever a remote client
	 *  requests to **write** the Published Audio Capabilities (PAC) location.
should be:
	 *  Write location callback is called whenever a remote client
	 *  requests to **read** the Published Audio Capabilities (PAC) location.
	 
PAC's are read only and my assumption was that the callback was called write as the application (PAC server) needs to write the value that the client wants to read?

Reply:
@Casper-Bonde-Bose No, the `write` description is correct. The Sink and Source location characteristics are (optionally) writeable: 
![image](https://user-images.githubusercontent.com/2520413/157852202-ca0985c7-1df6-4ad2-8486-af45343107ab.png)

Reply:
I see - Maybe a feature to be able to configure a speaker to be at a specific location?

Reply:
Possibly, yes. We (@asbjornsabo and I) raised this in the GAWG some time ago, and it seems that in hindsight it shouldn't have been writable.